### PR TITLE
Created a Basic LED Blinker Task for Demo Purposes.

### LINKED ISSUE #2: Create Basic Arduino Blink Sketch with FreeRTOS
Just to get things in the right place and to make sure everyone can build it.

### PR DESCRIPTION
**Pull request by gfarrand**
Fixes #2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,58 @@
-# knex-elevator
-Control a mutli-shaft elevator built from K'Nex.
+# KNEX Elevator
```
Comment (gfarrand) [inline README.md:1]: This file needs plenty of work, but I figure we can start filling this in as we get the system stood up more.

File: arduino/KnexElevator/BlinkerTask.cpp [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+////////////////////////////////////////////////////////////////////////////////
+void
+BlinkerTask::initialize(void)
+{
+    // Setup the debug LED for writing
+    pinMode(13, OUTPUT);
+    pinMode(2, INPUT_PULLUP);
+
+    // Fire up our task
+    xTaskCreate(BlinkerTask::runWrapper,  // Entry Point
```
Comment (gfarrand) [inline arduino/KnexElevator/BlinkerTask.cpp]: See the [FreeRTOS reference manual](http://www.freertos.org/Documentation/FreeRTOS_Reference_Manual_V9.0.0.pdf) for details on this and other functions in this file.

File: README.md (hunk 1 above)
Comment (saunmeht) [reply README.md:1]: Sounds good.

File: arduino/KnexElevator/BlinkerTask.h [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+/// @brief
+/// Abstracts the Smoke Alarm main task.
+////////////////////////////////////////////////////////////////////////////////
+class BlinkerTask
+{
+   public:
+      BlinkerTask(void);
+      ~BlinkerTask(void);
+
+      // To be called in setup()
```
Comment (saunmeht) [inline arduino/KnexElevator/BlinkerTask.h]: Add doxygen type comment block for the methods in the H-file

File: arduino/KnexElevator/BlinkerTask.cpp [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+                "Main",                   // Task Name
+                128,                      // Stack Depth
+                this,                     // Task Parameters
+                2,                        // Priority
+                NULL);                    // Storage for task handle
+}
+
```
Comment (saunmeht) [inline arduino/KnexElevator/BlinkerTask.cpp:72]: Remove this line.

File: arduino/KnexElevator/BlinkerTask.cpp [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+
+////////////////////////////////////////////////////////////////////////////////
+void
+BlinkerTask::runWrapper(void * pvParameters)
+{
+    // Entry point to the new task
+    BlinkerTask* p_this = reinterpret_cast<BlinkerTask*>(pvParameters);
```
Comment (saunmeht) [inline arduino/KnexElevator/BlinkerTask.cpp]: Camel case or Snake case, let's select one and run.

File: arduino/KnexElevator/BlinkerTask.cpp [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    p_this->run();
+}
+
```
Comment (saunmeht) [inline arduino/KnexElevator/BlinkerTask.cpp]: Add one more line.

File: arduino/KnexElevator/BlinkerTask.cpp [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    // Intentionally left empty
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+void
+BlinkerTask::initialize(void)
+{
+    // Setup the debug LED for writing
+    pinMode(13, OUTPUT);
```
Comment (saunmeht) [inline arduino/KnexElevator/BlinkerTask.cpp]: #define for the magic numbers

File: arduino/KnexElevator/BlinkerTask.cpp [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+// SOFTWARE.
+//
+// /////////////////////////////////////////////////////////////////////////////
+
+
+// /////////////////////////////////////////////////////////////////////////////
+// Include(s)
+// /////////////////////////////////////////////////////////////////////////////
+#include <Arduino_FreeRTOS.h>
+#include <Arduino.h>
```
Comment (saunmeht) [inline arduino/KnexElevator/BlinkerTask.cpp:32]: Add one blank line after this include.

File: arduino/KnexElevator/BlinkerTask.h [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+// /////////////////////////////////////////////////////////////////////////////
+// Include(s)
+// /////////////////////////////////////////////////////////////////////////////
+#include <Arduino_FreeRTOS.h>
+
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief
+/// Abstracts the Smoke Alarm main task.
```
Comment (gfarrand) [inline arduino/KnexElevator/BlinkerTask.h:38]: Oops, clean up copy-pasta.

File: arduino/KnexElevator/BlinkerTask.cpp [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+void
+BlinkerTask::initialize(void)
+{
+    // Setup the debug LED for writing
+    pinMode(13, OUTPUT);
+    pinMode(2, INPUT_PULLUP);
+
+    // Fire up our task
+    xTaskCreate(BlinkerTask::runWrapper,  // Entry Point
+                "Main",                   // Task Name
```
Comment (gfarrand) [inline arduino/KnexElevator/BlinkerTask.cpp]: Bad task name, fix.
__Update__: Fixed.

File: arduino/KnexElevator/BlinkerTask.h (hunk 9 above)
Comment (gfarrand) [inline arduino/KnexElevator/BlinkerTask.h:38]: This comment is helping me test/learn the "Start a Review" button.

File: arduino/KnexElevator/BlinkerTask.cpp (hunk 2 above)
Comment (roshroff) [reply arduino/KnexElevator/BlinkerTask.cpp]: This functions returns something to indicate if the task was created successfully or not. Do we want to have some sort of error handling here.

File: arduino/KnexElevator/BlinkerTask.cpp [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+                128,                      // Stack Depth
+                this,                     // Task Parameters
+                2,                        // Priority
+                NULL);                    // Storage for task handle
+}
+
+
+
```
Comment (roshroff) [inline arduino/KnexElevator/BlinkerTask.cpp]: Remove this line

File: arduino/KnexElevator/BlinkerTask.cpp (hunk 6 above)
Comment (roshroff) [inline arduino/KnexElevator/BlinkerTask.cpp]: Add a line if we are planning on keeping two line gap between two functions.

File: README.md [hunk 12]
```diff
@@ -1,2 +1,58 @@
...
+
+### Floor Marquee
+TODO: Talk about the 7-segment marquee display
+
+
+## Maintainers
+[@granfarr](https://github.com/granfarr).
+
+## Contribute
+[Open an issue](https://github.com/granfarr/knew-elevator/issues/new) or submit PRs.
```
Comment (DizzyBytes) [inline README.md]: Knew--> knex
Some issue with the link?

Comment (gfarrand) [reply README.md]: Yeah, that's a busted link for sure... will fix.

File: arduino/KnexElevator/BlinkerTask.cpp (hunk 7 above)
Comment (gfarrand) [reply arduino/KnexElevator/BlinkerTask.cpp]: Agreed, but this is going to get ripped out in the very near future. Once the other tasks get spun up, we will delete this blinker thread completely. Would like to leave as-is for the short term.

File: arduino/KnexElevator/BlinkerTask.cpp (hunk 4 above)
Comment (gfarrand) [reply arduino/KnexElevator/BlinkerTask.cpp:72]: Done.

File: arduino/KnexElevator/BlinkerTask.cpp (hunk 11 above)
Comment (gfarrand) [reply arduino/KnexElevator/BlinkerTask.cpp]: Done.

File: arduino/KnexElevator/BlinkerTask.cpp (hunk 5 above)
Comment (gfarrand) [reply arduino/KnexElevator/BlinkerTask.cpp]: I am all for the following:
1. PascalCase for names of classes, and methods (need update for method)
1. camelCase for arguments and stack variables

I will file an issue to get an AStyle config (or similar) into the repo and that way we can run the stylizer before committing.

File: arduino/KnexElevator/BlinkerTask.cpp (hunk 6 above)
Comment (gfarrand) [reply arduino/KnexElevator/BlinkerTask.cpp]: Okay - will settle on 2 lines throughout.

**Review by gfarrand — COMMENTED** [review]
@saunmeht - Please approve changes.

File: arduino/KnexElevator/BlinkerTask.cpp (hunk 5 above)
Comment (saunmeht) [reply arduino/KnexElevator/BlinkerTask.cpp]: I am fine with the suggestion.  Let's update this pull request and get that issue early in the backlog.

File: README.md [hunk 13]
```diff
@@ -1,2 +1,58 @@
...
+
+### Call Buttons
+TODO: Talk about call buttons
+
+### Floor Marquee
+TODO: Talk about the 7-segment marquee display
+
+
+## Maintainers
+[@granfarr](https://github.com/granfarr).
```
Comment (saunmeht) [inline README.md]: You can add us as maintainers/contributors as well. :)

Comment (gfarrand) [reply README.md]: Yeah, and spell my github user name correctly 😄

**Review by gfarrand — COMMENTED** [review]
@saunmeht - Ready for your approval.

**Review by saunmeht — APPROVED** [review]

**Pull request merged**